packages/eslint-plugin-office-addins/src/rules/call-sync-after-load.ts

```diff
@@ -79,21 +80,26 @@ export = {
               messageId: "callSyncAfterLoad",
               data: { name: node.name, loadValue: propertyName },
             });
+            usageDataObject.reportSuccess("call-sync-after-load");
           }
         }
       });
     }
 
     return {
       Program() {
-        apiReferences = findOfficeApiReferences(context.getScope());
-        apiReferences.sort((left, right) => {
-          return (
-            left.reference.identifier.range[1] -
-            right.reference.identifier.range[1]
-          );
-        });
-        findLoadBeforeSync();
+        try {
+          apiReferences = findOfficeApiReferences(context.getScope());
+          apiReferences.sort((left, right) => {
+            return (
+              left.reference.identifier.range[1] -
+              right.reference.identifier.range[1]
+            );
+          });
+          findLoadBeforeSync();
+        } catch(err: any) {
+          usageDataObject.reportException("call-sync-after-load", err);
+        }
```

Comment:
So if I understand correctly we are logging success after being able to report one instance of a customer code problem, an error if the rule code failed, and nothing if the rule code ran but no error in the user customer code was found.  What questions about customer behavior do we hope to answer with these data points?

Reply:
The main question we are trying to answer is:
"What are the most common mistakes customers make when writing their code with Office js API".
Checking the errors is more for us to catch any wrong implementation because we actually do not throw any errors, so it is more to check if our code might be faulty at some part.

Reply:
So what happens with live linting in VSCode . . . does the same rule trigger send the same telemetry until the customer fixes it?  We would need to make sure we aren't getting repeated signal on the same error.

Reply:
Yeah, this would lead to having more data overall, but I couldn't think of a way to make this not happen.

My idea was that customers either ignore or fix the error (then we wouldn't get the telemetry anymore). If they don't, then telemetry will keep getting sent.
I thought this wouldn't be that bad, given that the absolute number of times the rule was hit doesn't matter much. What really matters is the count between each rule.

Reply:
That would actually sku to someone who isn't fixing the error vs. someone who does.  It could be that something comes up as more common if a handful of people never fix the error vs. and error that people hit and fix.  Do we end up catering to people who aren't fixing flagged error vs. those who are finding the error reporting useful and fixing them.

I'm also wondering about the volume itself.  If the telemetry is getting logged during live linting we could be getting way too much signal in just one customer session.

It also just occurred to me that I'm not sure what happens to the telemetry if someone puts the comment in to ignore the rule (does the rule still continue to fire, silently hid the error, and telemetry get logged).

Reply:
I just tested with live linting and what happens is that the linter runs around every 10 seconds.
This means that we would get a lot, but really of a lot of noise data in a way that I don't quite think would be worth data.

I think now I will look into the telemetry of Office-Addin-Lint because this we can control we it gets called.